### ISSUE TITLE
Use bbox as visual prompts

### ISSUE THREAD

**Issue by PanchengZhao** [UX-Decoder/Segment-Everything-Everywhere-All-At-Once#161]

**Comment by PanchengZhao** [comment]
Hi, I ran the demo locally and it's a nice work!

I want to use this project as a utility to help me segment targets in specific areas. But I would like to know what to do with the input data to use the bbox as a visual cue.
I tried to draw the white bbox directly on the black base image as a mask input model but didn't get the expected results.

I'd appreciate it if there were samples of using bbox as input for reference

**Comment by MaeenBadea** [comment]
This is the hack I implemented, instead of looking for the src of the prompts I just overrided the masks in the for loop

```
def mask_to_box_with_perturb(mask: torch.Tensor, max_shift: int = 10) -> torch.Tensor:
    """
    Convert a binary mask tensor (H, W) to a bounding box mask with random perturbation.

    Args:
        mask (torch.Tensor): Binary tensor of shape (H, W).
        max_shift (int): Maximum random shift/expansion/shrink (in pixels).

    Returns:
        torch.Tensor: Binary tensor of same shape with perturbed bounding box.
    """
    # Get coordinates of foreground
    y, x = torch.where(mask > 0)

    if len(x) == 0 or len(y) == 0:
        # Empty mask -> return zeros
        return torch.zeros_like(mask, dtype=torch.uint8)

    H, W = mask.shape

    # Compute bounding box
    x_min, x_max = x.min().item(), x.max().item()
    y_min, y_max = y.min().item(), y.max().item()

    # Add random perturbation (shift / expand / shrink)
    dx1 = random.randint(-max_shift, max_shift)
    dx2 = random.randint(-max_shift, max_shift)
    dy1 = random.randint(-max_shift, max_shift)
    dy2 = random.randint(-max_shift, max_shift)

    x_min = max(0, x_min + dx1)
    x_max = min(W - 1, x_max + dx2)
    y_min = max(0, y_min + dy1)
    y_max = min(H - 1, y_max + dy2)

    # Create empty mask
    box_mask = torch.zeros_like(mask, dtype=torch.uint8, device=mask.device)

    # Fill perturbed bounding box
    box_mask[y_min:y_max+1, x_min:x_max+1] = 1

    return (box_mask > 0)
```

```

for i , b in enumerate(batch):
        for j, mask in enumerate(b['spatial_query']["gt_masks"]):
            bbox = mask_to_box_with_perturb(mask)
            b['spatial_query']['rand_shape'][j] = bbox                        
            b['spatial_query']['types'][j] = "circle"

```
this basically forms a binary tensor with 1s in the area covering the object of interest
although I'm not sure if the spatial_query type has any effect, but circle was the only type with filled masks that's why I set them up with this type
add this to both the training loop in default_trainer.py & evaluation step in XDecoderPipeline.py
